backend/lib/edgehog/triggers/handler/manual_actions/handle_trigger.ex

```diff
@@ -106,6 +109,72 @@ defmodule Edgehog.Triggers.Handler.ManualActions.HandleTrigger do
     |> Ash.create(tenant: tenant)
   end
 
+  defp handle_event(%IncomingData{interface: @available_images} = event, tenant, _realm_id, _device_id, _timestamp) do
+    case String.split(event.path, "/") do
+      ["", image_id, "pulled"] ->
+        with {:ok, containers} <- Containers.containers_with_image(image_id, tenant: tenant) do
+          releases =
+            containers
+            |> Enum.reduce_while([], &releases_from_container(&1, &2, tenant))
+            |> Enum.uniq()
+
+          deployments =
+            releases
+            |> Enum.reduce_while([], &deployments_from_release(&1, &2, tenant))
+            |> Enum.uniq()
+
+          Enum.map(deployments, &Containers.deployment_update_satus(&1))
+        end
+
+      _ ->
+        :ok
+    end
+  end
+
+  defp handle_event(%IncomingData{interface: @available_networks} = event, tenant, _realm_id, _device_id, _timestamp) do
+    case String.split(event.path, "/") do
+      ["", network_id, "pulled"] ->
+        with {:ok, relations} <- Containers.containers_with_network(network_id, tenant: tenant),
+             {:ok, relations} <- Ash.load(relations, [:container]) do
+          containers = relations |> Enum.map(& &1.container) |> Enum.uniq()
+
+          releases =
+            containers
+            |> Enum.reduce_while([], &releases_from_container(&1, &2, tenant))
+            |> Enum.uniq()
```

Comment:
Even if something errors out, I'd say we want to update the status of the containers we were able to gather. And possibly logging the error:

```suggestion
          releases =
            containers
            |> Enum.map(&Containers.releases_with_container(&1.id, tenant: tenant))
            |> Enum.flat_map(fn
              {:ok, releases} -> releases
              {:error, reason} -> 
                Logger.error("Error reading releases from containers: #{inspect(reason)}")
                []
            end)
            |> Enum.uniq()
```